x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/util/SourceUtils.java

```diff
@@ -21,73 +19,17 @@ public final class SourceUtils {
     private SourceUtils() {}
 
     public static void writeSource(StreamOutput out, Source source) throws IOException {
-        writeSource(out, source, true);
-    }
-
-    public static void writeSourceNoText(StreamOutput out, Source source) throws IOException {
-        writeSource(out, source, false);
-    }
-
-    public static Source readSource(StreamInput in) throws IOException {
-        return readSource(in, null);
-    }
-
-    public static Source readSourceWithText(StreamInput in, String queryText) throws IOException {
-        return readSource(in, queryText);
-    }
-
-    private static void writeSource(StreamOutput out, Source source, boolean writeText) throws IOException {
         out.writeInt(source.source().getLineNumber());
         out.writeInt(source.source().getColumnNumber());
-        if (writeText) {
-            out.writeString(source.text());
-        } else {
-            out.writeInt(source.text().length());
-        }
+        out.writeString(source.text());
     }
 
-    private static Source readSource(StreamInput in, @Nullable String queryText) throws IOException {
+    public static Source readSource(StreamInput in) throws IOException {
         int line = in.readInt();
         int column = in.readInt();
         int charPositionInLine = column - 1;
 
-        String text;
-        if (queryText == null) {
-            text = in.readString();
-        } else {
-            int length = in.readInt();
-            text = sourceText(queryText, line, column, length);
-        }
+        String text = in.readString();
         return new Source(new Location(line, charPositionInLine), text);
```

Comment:
I've kept this second serialization mechanism in place for now. It should probably be reworked so we don't use it any more, but that's going to take some more time.